docs/userguide/searchspace.md

```diff
@@ -1,46 +1,269 @@
+~~~python
+searchspace = SearchSpace(discrete=discrete_subspace, continuous=continuous_subspace)
+~~~
 
-A discrete/continuous search space is a s searchspace that was constructed by only providing a discrete/continuous subspace upon initialization.
+While this constructor is the default choice, it might not be the most convenient.
+Consequently, other constructors are available.
 
-### Discrete subspaces
+### Building from the Product of Parameter Values
 
-In addition to the ones noted above, a discrete subspace has the following attributes:
-* **The experimental representation:** A ``DataFrame`` representing the experimental representation of the subspace.
-* **The metadata:** A ``DataFrame`` keeping track of different metadata that is relevant for running a campaign.
-* **An "empty" encoding flag:** A flag denoting whether an "empty" encoding should be used. This is useful, for instance, in combination with random recommenders that do not read the actual parameter values.
-* **The computational representation:** The computational representation of the space. If not provided explicitly, it will be derived from the experimental representation.
+The function [`SearchSpace.from_product`](baybe.searchspace.core.SearchSpace.from_product) is analog to the corresponding function available for `SubspaceDiscrete`, but allows the parameter list to contain both discrete and continuous parameters.
 
-Although it is possible to directly create a discrete subspace via the ``__init__`` function, it is intended to create themvia the [`from_dataframe`](baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe) or [`from_product`](baybe.searchspace.discrete.SubspaceDiscrete.from_product) methods. These methods either require a ``DataFrame`` containing the experimental representation of the parameters and the optional explicit list of parameters (``from_dataframe``) or a list of parameters and optional constraints (``from_product``).
+### Constructing from a dataframe
 
-For details and examples on how to use a discrete search space, see the corresponding example [here](./../../examples/Searchspaces/discrete_space) and [here](./../../examples/Constraints_Discrete/Constraints_Discrete).
+[`SearchSpace.from_dataframe`](baybe.searchspace.core.SearchSpace.from_dataframe) constructs a search space from a given dataframe.
+Due to the ambiguity between discrete and numerical parameter choices, this function requires a parameters list with pre-defined parameter objects, unlike its subspace counterparts, where this list is optional.
 
-### Continuous subspaces
+```python
+params = [
+    NumericalDiscreteParameter(name="x0", values=[1, 2, 3]),
+    NumericalDiscreteParameter(name="x1", values=[4, 5, 6]),
```

Comment:
it would be a slightly more interesting example if these would not just all be discrete numerical ones but at least one   continuous one

also in the print below showing then the complete pretty print of the hybrid space would make sense